memory/StackMap.h

```diff
@@ -0,0 +1,302 @@
+        }
+
+        return std::make_pair(end(), false); // full
+    }
+
+    std::pair<iterator, bool> emplace(const KeyT& key, const T& value) { return add(key, value); }
+
+    bool erase(const KeyT& key)
+    {
+        auto keyHash = utils::hash<KeyT>{}(key);
+        for (uint32_t i = 0; i < _maxSpread; ++i)
+        {
+            auto& indexEntry = _index[indexPosition(keyHash, i)];
+            if (indexEntry.keyHash == keyHash && indexEntry.position)
+            {
+                _elements[indexEntry.position - 1] = ElementEntry();
+                indexEntry = IndexEntry();
+                --_count;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    const T& operator[](const KeyT& key) const
+    {
+        auto keyHash = utils::hash<KeyT>{}(key);
+        for (uint32_t i = 0; i < _maxSpread; ++i)
+        {
+            const auto& indexEntry = _index[indexPosition(keyHash, i)];
+            if (indexEntry.keyHash == keyHash && indexEntry.position)
+            {
+                return _elements[indexEntry.position - 1].keyValue.second;
+            }
+        }
+
+        return _emptyObject;
+    }
+
+    T& operator[](const KeyT& key)
+    {
+        auto keyHash = utils::hash<KeyT>{}(key);
+        for (uint32_t i = 0; i < _maxSpread; ++i)
+        {
+            const auto& indexEntry = _index[indexPosition(keyHash, i)];
+            if (indexEntry.keyHash == keyHash && indexEntry.position)
+            {
+                return _elements[indexEntry.position - 1].keyValue.second;
+            }
+        }
+
+        if (add(key, T()).second)
+        {
+            return (*this)[key];
+        }
+        return _emptyObject;
+    }
+
+    iterator find(const KeyT& key)
+    {
+        const auto& map = *this;
+        return iterator(map.find(key));
+    }
+
+    const_iterator find(const KeyT& key) const
+    {
+        auto keyHash = utils::hash<KeyT>{}(key);
+        for (uint32_t i = 0; i < _maxSpread; ++i)
+        {
+            const auto& indexEntry = _index[indexPosition(keyHash, i)];
+
+            if (indexEntry.keyHash == keyHash && indexEntry.position)
+            {
+                return const_iterator(const_cast<ElementEntry*>(_elements.data()), indexEntry.position - 1, _end);
+            }
+        }
+        return end();
+    }
+
+    bool contains(const KeyT& key) const
+    {
+        auto keyHash = utils::hash<KeyT>{}(key);
+        for (uint32_t i = 0; i < _maxSpread; ++i)
+        {
+            const auto& indexEntry = _index[indexPosition(keyHash, i)];
+
+            if (indexEntry.keyHash == keyHash && indexEntry.position)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    size_t capacity() const { return _elements.size(); }
+    size_t size() const { return _count; }
+    bool empty() const { return _count == 0; }
```

Comment:
It must remove all `commited` flags and `indexEntry.position`.

Reply:
you are thinking about clear method

Reply:
Yes, wrong line :)

Reply:
Yeah sorry, never mind. I guess I misinterpreted the code as `_count = 0` and my brain auto renamed this method to "clean". I am seeing now there is no clean method implemented. You can forget the comment